GetStream/stream-chat-android
# Fix/socket service fix

### 🎯 Goal

We want to keep possibility to connect to socket when user was already set (in case of the push messages flow). For that we need to change logic of `connectUser` method

### 🛠 Implementation details

1. Change logic of connect user according to `UserState`, `SocketState`
2. Cover this method by tests

### 🧪 Testing

1. Receive push
2. Click on notification asap
3. Check app is working fine

### ☑️ Checklist


### 🎉 GIF

![](https://media4.giphy.com/media/zPbnEgxsPJOJSD3qfr/giphy.gif)
